implementations/rust/ockam/ockam_core/src/vault/types.rs

```diff
@@ -183,11 +183,11 @@ impl From<Signature> for SignatureVec {
     }
 }
 
-/// All possible [`SecretType`]s
+/// All possible [`KeyType`]s
 #[derive(Serialize, Deserialize, Copy, Clone, Debug, Encode, Decode, Eq, PartialEq, Zeroize)]
 #[rustfmt::skip]
 #[cbor(index_only)]
-pub enum SecretType {
+pub enum KeyType {
```

Comment:
I'm not sure renaming Secret -> Key is technically correct. Key is more narrow term, for example I don't think a Buffer with sensitive data can be called a Key

Reply:
Yes and that might be a hint that we should separate this data from the rest. For example in `hasher_impl.rs` we try to retrieve the salt and [fail at runtime if it doesn't have the right type](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_vault/src/hasher_impl.rs#L52). Maybe this is an indication that this kind of data (that is always `Ephemeral` as far as I understand) should be stored separately inside a `Vault`.